core/trino-main/src/main/java/io/trino/execution/scheduler/FixedBucketNodeMap.java

```diff
@@ -47,6 +48,12 @@ public int getBucketCount()
         return bucketToNode.size();
     }
 
+    @Override
+    public int getNodeCount()
+    {
+        return ImmutableSet.copyOf(bucketToNode).size();
```

Comment:
This is to find a number of unique nodes. The immutability doesn't have any special meaning here, just a useful constructor method for a set. Ideally it should be refactored to provide a number of unique nodes explicitly, as a constructor parameter. But the refactor wasn't trivial. I decided to delay the refactor until later since the impact of this de-duplication is not that high (it is only called once per stage and fixed mappings is rather a corner case and are rare).

Reply:
This would be more idiomatic for that purpose:

```java
bucketToNode.stream()
    .distinct()
    .count();
```